extensions/amp-form/0.1/amp-form.js

```diff
@@ -94,12 +102,16 @@ export class AmpForm {
    * @private
    */
   handleSubmit_(e) {
-    if (e.defaultPrevented) {
+    if (this.state_ == FormState_.SUBMITTING) {
+      e.preventDefault();
       return;
     }
 
-    if (this.state_ == FormState_.SUBMITTING) {
+    const shouldValidate = !this.form_.hasAttribute('novalidate');
+    if (shouldValidate &&
+        this.form_.checkValidity && !this.form_.checkValidity()) {
       e.preventDefault();
+      this.vsync_.run({mutate: reportValidity}, {form: this.form_});
```

Comment:
You'll need to use the same `{measure: undefined, mutate: reportValidity}` hidden class. 😞

I'll file an issue to add `state` to `#measure` and `#mutate`.